### PR TITLE
feat(item-edit): support lenient best-effort partial edits

### PR DESCRIPTION
**Pull request by urjitc**
## Summary
- add opencode-style safe fallback matchers to item edit replacement flow (exact + whitespace/context tolerant strategies)
- introduce `applyEditsBestEffort` to apply successful edits while collecting failed edit diagnostics
- update document/quiz/flashcard worker edit paths to persist partial successes and return `success=false` with `appliedEditIndices` and `failedEdits`
- keep tool description concise while documenting partial-apply behavior
- add coverage for Satisfaction-style whitespace mismatch, ambiguity handling, and partial-apply diagnostics

## Test plan
- [x] `pnpm vitest "src/lib/utils/__tests__/edit-replace.test.ts" "src/lib/ai/tools/__tests__/edit-item-tool.test.ts"`
- [x] `ReadLints` on edited files (no linter errors)

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **Bug Fixes**
  * Edit application no longer fully rolls back on some failures; partial edits are applied when possible and reported.

* **New Features**
  * Improved matching that tolerates whitespace/context fallbacks for more reliable edits.
  * Detailed failure diagnostics now report which edits succeeded, which failed, and why.

* **Tests**
  * Added coverage for partial-apply scenarios, best-effort edit behavior, and matching edge cases.

### CODE REVIEW

File: src/lib/utils/edit-replace.ts [hunk 1]
```diff
@@ -97,6 +108,234 @@ function countOccurrences(content: string, oldText: string): number {
...
+        if (blockLine.length > 0 || findLine.length > 0) {
+          totalNonEmptyLines++;
+          if (blockLine === findLine) matchingLines++;
+        }
+      }
+      if (totalNonEmptyLines === 0 || matchingLines / totalNonEmptyLines >= 0.5) {
+        yield blockLines.join("\n");
+        break;
+      }
+      break;
```
[the remark this answers is not in the document]
Comment (urjitc) [reply src/lib/utils/edit-replace.ts]: Fixed in `1e78b779`: raised the `contextAwareReplacer` interior-line similarity threshold from 0.5 to 0.8 and added a regression test that rejects weak 50% matches.

File: src/lib/utils/edit-replace.ts [hunk 2]
```diff
@@ -186,6 +399,131 @@ export function applyEdits(content: string, edits: Edit[]): AppliedEditsResult {
...
+        index: curr.editIndex,
+        reason: `edits[${prev.editIndex}] and edits[${curr.editIndex}] overlap. Merge them into one edit or target disjoint regions.`,
+      });
+      continue;
+    }
+    nonOverlapping.push(curr);
+  }
+
+  let currentContent = applyResolvedEdits(normalizedContent, nonOverlapping);
+  for (const resolved of nonOverlapping) appliedEditIndices.push(resolved.editIndex);
```
[the remark this answers is not in the document]
Comment (urjitc) [reply src/lib/utils/edit-replace.ts:478]: Fixed in `1e78b779`: overlap-rejected edits are now re-queued into `pending` for pass 2 retry (instead of immediate failure), with test coverage for the retry path.

**Comment by urjitc** [comment]
Addressed the review findings in commit 1e78b779.

- Added `itemName` to partial-apply (`partialApplied`) tool responses in `edit-item-tool`.
- Tightened `contextAwareReplacer` similarity threshold to `0.8` and added a regression test for weak 50% matches.
- Re-queued overlap-rejected pass-1 edits for pass-2 retry, plus pass-2 retry coverage tests.
- Expanded `applyEditsBestEffort` tests for all-success and all-fail scenarios.

Validated with:
- `pnpm test src/lib/utils/__tests__/edit-replace.test.ts src/lib/ai/tools/__tests__/edit-item-tool.test.ts`
